lib/redis/cluster.rb

```diff
@@ -238,6 +239,36 @@ def try_send(node, method_name, *args, retry_count: 3, &block)
       raise
     end
 
+    def _scan(command, &block)
+      input_cursor = command[1].to_i
+
+      client_index = input_cursor % 2**8
+      raw_cursor = input_cursor >> 8
+
+      clients = @node
```

Comment:
It might be readable if this step is added as a public method to `Redis::Cluster::Node` class.

```ruby
class Redis
  class Cluster
    class Node
      def scale_reading_clients
        @clients.select { |k, v| replica_disabled? ? master?(k) : slave?(k) }.values
      end
    end
  end
end
```